src/RippleKey.cpp

```diff
@@ -187,6 +197,14 @@ RippleKey::multiSign(ripple::STTx& tx) const
     {
         return (a[sfAccount] < b[sfAccount]);
     });
+
+    // Re-serialize this signed and sorted STTx so the hash is freshly computed.
+    Serializer s2;
+    tx->add (s2);
+    Blob txBlob = s2.getData ();
+    SerialIter sit {makeSlice(txBlob)};
+    std::shared_ptr<STTx const> txNew = std::make_shared<STTx const> (sit);
+    tx.emplace (*txNew);
```

Comment:
Thanks for the suggestion to skip the `make_shared`.  Much cleaner.

Regarding the question, yup.  The problem is that you can't assign to an `STTx`.  And the `STTx` caches its transaction ID.  And there are only a few specific circumstances where the cached transaction ID is updated.  The easiest way to make sure the transaction ID is correct is to make a new `STTx`, which is what we're doing here.  And since you can't assign to the `STTx`, we have to replace it wholesale.  Hence the `optional`.

I started to add a new method to the `STTx` which would allow me to force it to recompute the transaction ID.  But then I would need to wait for that change to propagate through ripple-libpp.  The technique used here is the same technique that multisigning uses, so I figure it's good enough.  A small loss in efficiency will not kill us here.